### PR TITLE
[OpenLR] Fix matching evaluation tool.

### PR DESCRIPTION
**Pull request by mgsergio**

### CODE REVIEW

File: tools/python/openlr/quality.py [hunk 1]
```diff
@@ -97,6 +97,9 @@ def almost_equal(s1, s2, eps=1e-5):
     )
 
 def common_part(l1, l2):
+    assert l1
```
Comment (burivuh) [inline tools/python/openlr/quality.py]: maybe a human-readable message with an Exception?

File: tools/python/openlr/quality.py [hunk 2]
```diff
@@ -143,32 +149,38 @@ def parse_segments(tree, limit):
         # within limit are considered accurate, so golden path should be equal
         # matched path.
         golden_route = parse_route(s.find('GoldenRoute')) or matched_route
-        if not matched_route and not golden_route:
-            raise ValueError('At least one of golden route or matched route should be present')
+        yield Segment(segment_id, golden_route, matched_route)
```
Comment (burivuh) [inline tools/python/openlr/quality.py]: In the this version param order has changed?

File: tools/python/openlr/quality.py [hunk 3]
```diff
@@ -143,32 +149,38 @@ def parse_segments(tree, limit):
...
+
+def calculate(tree):
+    result = {}
+    for s in parse_segments(tree, args.limit):
         try:
-            yield Segment(segment_id, matched_route, golden_route)
+            result[s.segment_id] = common_part(s.golden_route, s.matched_route)
         except:
```
Comment (burivuh) [inline tools/python/openlr/quality.py]: Please describe exception and what can go wrong

File: tools/python/openlr/quality.py [hunk 4]
```diff
@@ -143,32 +149,38 @@ def parse_segments(tree, limit):
...
+    return result
 
 def merge(src, dst):
-    golden_routes = {
-        int(s.find('.//ReportSegmentID').text) : s.find('GoldenRoute')
-        for s in src.findall('Segment')
-    }
+    golden_routes = dict(
+        filter(
+            lambda x: x[1],
```
Comment (burivuh) [inline tools/python/openlr/quality.py]: Just if clause of the dict comprehension?

File: tools/python/openlr/quality.py [hunk 5]
```diff
@@ -143,32 +149,38 @@ def parse_segments(tree, limit):
...
-    }
+    return result
 
 def merge(src, dst):
-    golden_routes = {
-        int(s.find('.//ReportSegmentID').text) : s.find('GoldenRoute')
-        for s in src.findall('Segment')
-    }
+    golden_routes = dict(
+        filter(
```
Comment (burivuh) [inline tools/python/openlr/quality.py]: I think check for emptiness later is more readable than this filter

File: tools/python/openlr/quality.py (hunk 1 above)
Comment (mgsergio) [reply tools/python/openlr/quality.py]: Done. A Message was added.

File: tools/python/openlr/quality.py (hunk 2 above)
Comment (mgsergio) [reply tools/python/openlr/quality.py]: Yep

File: tools/python/openlr/quality.py (hunk 3 above)
Comment (mgsergio) [reply tools/python/openlr/quality.py]: Done.

File: tools/python/openlr/quality.py (hunk 5 above)
Comment (mgsergio) [reply tools/python/openlr/quality.py]: Done.

File: tools/python/openlr/quality.py [hunk 6]
```diff
@@ -143,33 +148,36 @@ def parse_segments(tree, limit):
...
-        for s in parse_segments(tree, args.limit)
-    }
+    result = {}
+    for s in parse_segments(tree, args.limit):
+        try:
+            result[s.segment_id] = common_part(s.golden_route, s.matched_route)
+        except AssertionError:
+            print('Something is wrong with segment {}'.format(s))
+            raise
+        except ValueError:
```
Comment (burivuh) [inline tools/python/openlr/quality.py]: Please describe the problem and maybe wrap in a new Exception with a clear message?

Comment (mgsergio) [reply tools/python/openlr/quality.py]: Done.

**Pull request merged**